### PR TITLE
[C/PyTorch] Add workspace optimization for fused attention arbitrary_seqlen backend

### PR DESCRIPTION
**Pull request by cyanguwa**
This PR:

- adds workspace optimizations for fused attention arbitrary_seqlen backend in cuDNN 8.9.5+
- changes the default of `NVTE_FUSED_ATTN_USE_FAv2_BWD` to 0 as this is the majority of the use cases
- moves zero fill algorithms to FP8 only (other backends do not need zero initialization)

### CODE REVIEW

**Comment by cyanguwa** [comment]
Please hold off on the merge

File: transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu [hunk 1]
```diff
@@ -1298,13 +1325,44 @@ void fused_attn_arbitrary_seqlen_bwd_qkvpacked(size_t batch, size_t max_seqlen,
...
+                int dp_workspace_limit = std::stoi(env_dp_workspace_limit);
+                if (dp_workspace_limit > max_allowed_dp_workspace) {
+                    max_allowed_dp_workspace = dp_workspace_limit;
+                }
+            } catch (...) {
+                NVTE_ERROR(
+                "Invalid argument for NVTE_FUSED_ATTN_DP_WORKSPACE_LIMIT (integer; in MBytes)! \n");
+            }
+        }
+        if (required_dp_workspace < max_allowed_dp_workspace) {
```
Comment (zlsh80826) [inline transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu:1353]: Is this `<` or `<=`? The current code seems not enable `use_workspace_opt` when  required_dp_workspace exactly equal to 256MB.

Comment (cyanguwa) [reply transformer_engine/common/fused_attn/fused_attn_f16_arbitrary_seqlen.cu:1353]: I can add the `=` :).

**Comment by cyanguwa** [comment]
Pipeline 9759553

**Review by zlsh80826 — APPROVED** [review]

**Pull request merged**